GithubRepo/Data/Repository/GithubRepository.swift

```diff
@@ -0,0 +1,23 @@
+//
+//  GithubRepository.swift
+//  GithubRepo
+//
+//  Created by Marcos Alves on 15/10/21.
+//
+
+import Foundation
+
+protocol GithubRepositoryDelegate: AnyObject {
+    func didChangeLoading()
```

Comment:
You can pass the new loading state as the parameter here. So you gonna have direct access for the new value.
Also, for delegates we usually pass the object as parameter as well. For instance, the table view delegates always pass the table view object in the parameters list.

```suggestion
    func githubRepositoryDidChangeLoading(_ githubRepository: GithubRepository, loading: Bool)
```